pandas/tests/test_resample.py

```diff
@@ -3103,6 +3103,21 @@ def f(x):
         result = g.apply(f)
         assert_frame_equal(result, expected)
 
+    def test_apply_with_mutated_index(self):
+        # GH 15169
+        index = pd.date_range('1-1-2015', '12-31-15', freq='D')
+        df = pd.DataFrame(data={'col1': np.random.rand(len(index))},
+                          index=index)
+
+        def f(x):
+            s = pd.Series([1, 2], index=['a', 'b'])
+            return s
+
+        expected = df.groupby(pd.Grouper(freq='M')).apply(f)
+
+        result = df.resample('M').apply(f)
+        assert_frame_equal(result, expected)
```

Comment:
can you add a test with a Series as well. (as you now have a separate case for that).

Reply:
@jreback 

Added. Let me know if the test is okay or not.